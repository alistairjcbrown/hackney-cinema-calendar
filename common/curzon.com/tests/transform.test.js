const transform = require("../transform");
const moviesWithLocalScreenings = require("./mock-data/movie-with-local-screenings.json");

const formatParameter = (result) => result;
const aldgate = {
  domain: "https://www.curzon.com",
  url: "https://www.curzon.com/venues/aldgate",
};

jest.useFakeTimers().setSystemTime(new Date("2025-01-11"));

describe("Curzon cinema (common)", () => {
  describe("when no movies provided", () => {
    it("returns an empty list of events", async () => {
      const value = formatParameter([]);
      expect(await transform(aldgate, value, {})).toEqual([]);
    });
  });

  describe("when movies with local screenings provided for Aldgate", () => {
    it("returns a list of events", async () => {
      const value = formatParameter(moviesWithLocalScreenings);
      expect(await transform(aldgate, value, {})).toMatchSnapshot();
    });
  });
});
