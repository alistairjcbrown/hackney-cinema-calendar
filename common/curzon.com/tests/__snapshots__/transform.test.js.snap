// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curzon cinema (common) when movies with local screenings provided for Aldgate returns a list of events 1`] = `
[
  {
    "overview": {
      "actors": [
        "Connie Nielsen",
        "Derek Jacobi",
        "Paul Mescal",
        "Pedro Pascal",
        "Joseph Quinn",
        "Fred Hechinger",
        "Lior Raz",
        "Denzel Washington",
      ],
      "categories": [
        "Drama",
      ],
      "certification": "15",
      "directors": [
        "Ridley Scott",
      ],
      "duration": 8880000,
      "trailer": "https://www.youtube.com/watch?v=3_OFWynGGuU",
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53872/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736595600000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53877/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736680800000,
      },
    ],
    "title": "Gladiator II",
    "url": "https://www.curzon.com/films/gladiator-ii/HO00005629/?siteId=ALD1",
  },
  {
    "overview": {
      "actors": [
        "Aunjanue Ellis",
        "Daveed Diggs",
        "Fred Hechinger",
        "Hamish Linklater",
        "Robert Aberdeen",
        "Ethan Herisse",
        "Brandon Wilson",
      ],
      "categories": [
        "Drama",
      ],
      "certification": "12A",
      "directors": [
        "RaMell Ross",
      ],
      "duration": 8400000,
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53867/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736605800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53866/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736626800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53880/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736691300000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53879/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736712000000,
      },
    ],
    "title": "Nickel Boys",
    "url": "https://www.curzon.com/films/nickel-boys/HO00005823/?siteId=ALD1",
  },
  {
    "overview": {
      "actors": [
        "Simon McBurney",
        "Nicholas Hoult",
        "Ralph Ineson",
        "Willem Dafoe",
        "Aaron Taylor-Johnson",
        "Emma Corrin",
        "Bill Skarsg√•rd",
        "Lily-Rose Depp",
      ],
      "categories": [
        "Drama",
        "Fantasy",
        "Horror",
      ],
      "certification": "15",
      "directors": [
        "Robert Eggers",
      ],
      "duration": 7920000,
      "trailer": "https://www.youtube.com/watch?v=Px6S0RxfAHg",
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53933/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736598000000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53869/",
        "notes": "Closed Captioned: Closed Captioned screening for Hard of Hearing",
        "screen": "Screen 1",
        "time": 1736607600000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53682/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736618400000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53681/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736628000000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53934/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736683200000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53882/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736694000000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53875/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736704800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53874/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736713200000,
      },
    ],
    "title": "Nosferatu",
    "url": "https://www.curzon.com/films/nosferatu/HO00005839/?siteId=ALD1",
  },
  {
    "overview": {
      "actors": [
        "Florence Pugh",
        "Marama Corlett",
        "Adam James",
        "Andrew Garfield",
      ],
      "categories": [
        "Drama",
        "Romance",
      ],
      "certification": "15",
      "directors": [
        "John Crowley",
      ],
      "duration": 6420000,
      "trailer": "https://www.youtube.com/watch?v=bS0Clau5700",
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53865/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736596800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53868/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736617200000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53878/",
        "notes": "Dog Days: Weekly screenings with lowered sound and complimentary Naturally Meaty Treats from Butcher's for dog owners and their canine companions. (Dog lovers without one of their own are also welcome.)",
        "screen": "Screen 4",
        "time": 1736682000000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53881/",
        "notes": "",
        "screen": "Screen 4",
        "time": 1736702400000,
      },
    ],
    "title": "We Live in Time",
    "url": "https://www.curzon.com/films/we-live-in-time/HO00005846/?siteId=ALD1",
  },
  {
    "overview": {
      "actors": [
        "Nicholas Hoult",
        "Jude Law",
        "Marc Maron",
        "Tye Sheridan",
        "Jurnee Smollett",
      ],
      "categories": [
        "Drama",
        "Crime",
      ],
      "certification": "15",
      "directors": [
        "Justin Kurzel",
      ],
      "duration": 6960000,
      "trailer": "https://www.youtube.com/watch?v=r7vwsrUlFxY",
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53871/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736608800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53884/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736692800000,
      },
    ],
    "title": "The Order",
    "url": "https://www.curzon.com/films/the-order/HO00005851/?siteId=ALD1",
  },
  {
    "overview": {
      "actors": [
        "Antonio Banderas",
        "Nicole Kidman",
        "Harris Dickinson",
        "Sophie Wilde",
        "Jean Reno",
        "Anoop Desai",
      ],
      "categories": [
        "Drama",
        "Romance",
      ],
      "certification": "18",
      "directors": [
        "Halina Reijn",
      ],
      "duration": 6840000,
      "trailer": "https://www.youtube.com/watch?v=dq_bnfZy9-g",
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53648/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736602800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53646/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736620800000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53679/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736629200000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53661/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736678400000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53663/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736696400000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53876/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736703600000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53660/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736714400000,
      },
    ],
    "title": "Babygirl",
    "url": "https://www.curzon.com/films/babygirl/HO00005853/?siteId=ALD1",
  },
  {
    "overview": {
      "actors": [
        "Will Sharpe",
        "Jesse Eisenberg",
        "Jennifer Grey",
        "Kieran Culkin",
        "Kurt Egyiawan",
        "Liza Sadovy",
        "Daniel Oreskes",
      ],
      "categories": [
        "Drama",
        "Comedy",
      ],
      "certification": "15",
      "directors": [
        "Jesse Eisenberg",
      ],
      "duration": 5400000,
      "trailer": "https://www.youtube.com/watch?v=lGeZQWMMDgM",
    },
    "performances": [
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53649/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736594400000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53647/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736612400000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53680/",
        "notes": "",
        "screen": "Screen 1",
        "time": 1736619600000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53645/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736630400000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53662/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736688000000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53664/",
        "notes": "",
        "screen": "Screen 2",
        "time": 1736706000000,
      },
      {
        "bookingUrl": "https://www.curzon.com/ticketing/seats/ALD1-53873/",
        "notes": "",
        "screen": "Screen 3",
        "time": 1736715600000,
      },
    ],
    "title": "A Real Pain",
    "url": "https://www.curzon.com/films/a-real-pain/HO00005873/?siteId=ALD1",
  },
]
`;
